packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -84,7 +84,7 @@ export const PanelsWrapper = React.forwardRef((props, forwardedRef) => {
       ReactDOM.flushSync(() => setActivePanelId(newActiveId));
       onActiveIdChange.current?.(newActiveId);
 
-      ref.current?.ownerDocument.getElementById(newActiveId)?.scrollIntoView({
+      ref.current?.querySelector(`[id='${newActiveId}']`)?.scrollIntoView({
```

Comment:
Have you considered using [`getRootNode()`](https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode) to grab the owner document or shadow-root, and then calling `getElementById` on it?

e.g. something like:

```suggestion
      ref.current?.getRootNode()?.querySelector(newActiveId)?.scrollIntoView({
```

Reply:
nit: You can type it as `Document | ShadowRoot`. Type assertions are all lies, so it doesn't matter for the purposes of the runtime code, but it's more accurate and descriptive (since the issue we are trying to solve is specifically related to shadow-roots).

The third possible return value of `getRootNode()` does make me nervous, but hopefully we never run into it 🤞

Reply:
I already mentioned the reasoning: it's more descriptive. The whole point of using `getRootNode()` is so that we get the encompassing shadow-root (if there is one). Think of it like a code comment.